pkg/model/mask.go

```diff
@@ -135,3 +157,17 @@ func UpdateService(src, dst *Service, mask Mask) error {
 
 	return nil
 }
+
+// UpdateTranslation updates the destination translation based on the source translation and field mask.
+func UpdateTranslation(src, dst *Translation, mask Mask) error {
```

Comment:
I would go with following logic, that would simplify everything

1. Check mask, if it does not contain messages, then just call update()
2. If it exists, then implement logic of updating slice here, not in update() function. Pop the messages from mask, and call update, if mask is not empty yet

Reply:
We should ignore managing read-only fields for now and determine general-purpose and reusable solutions later.

It should be closer to the request validation. Events (gRPC etc) that have reached inside our system MUST be already safe to process.

```go
// allowMask returns true if updateMask does not contain any readOnlyMask.
func allowMask(updateMask, readOnlyMask Mask) bool
```

Reply:
there is no mention of read-only fields in my comment

Reply:
We don't need to define `func Update<resource>` for each resource. We can get away with general-purpose functions.

Reply:
We don't need `UpdateTranslation` anymore.

Let's expose `update` and use it instead.